### ISSUE TITLE
Deletion of datasets in indexer via webhook isn't working

### ISSUE THREAD

**Issue by AlexGilleran** [TerriaJS/magda#1098]
### Problem description
Every time a data source is crawled, the crawler deletes old datasets - this should then propagate to the indexer via webhook. However the last manual reindex of the prod index looked like this:

```
[INFO] [05/28/2018 07:04:28.118] [indexer-akka.actor.default-dispatcher-3176] [RegistryCrawler(akka://indexer)] Indexed 57814 datasets with 0 failures
[INFO] [05/28/2018 07:04:28.118] [indexer-akka.actor.default-dispatcher-3176] [RegistryCrawler(akka://indexer)] Trimming datasets indexed before 2018-05-28T06:44:46.185Z
[INFO] [05/28/2018 07:04:28.961] [indexer-akka.actor.default-dispatcher-3176] [akka.actor.ActorSystemImpl(indexer)] Trimmed 6028 old datasets
```

_6028_ old datasets that needed trimming. Evidently the indexer isn't getting some or all of the deletion events anymore, or the deletion events aren't being created in the first place. It's clear that the datasets are actually being deleted in the registry though, otherwise the indexer wouldn't even know to trim them.

**Comment by t83714** [comment]
I think this trimming process (triggered by full crawl) is probably initiated inside indexer.
indexer just simply deletes any records in elasticsearch that has an old date (older than full crawl start time) in `indexed` field as only records not available from the registry will miss the `indexed` field update. There should be no events need to be generated.

**Comment by AlexGilleran** [comment]
Yeah but it should happen in real time rather than requiring a full recrawl :(.

**Comment by AlexGilleran** [comment]
To be clear, every time something gets deleted in the registry it should generate a delete event.

The indexer should be listening to these events, and should be deleting the corresponding datasets from its index when it gets them.

So the reindex trim should only ever trim a few datasets, because the webhooks hadn't caught up yet.

**Comment by t83714** [comment]
Oh I see. When do we normally need to delete records? 
Probably when connectors finished the data fetching?
If delete events are not created properly, the issue must be around here, right? @AlexGilleran

**Comment by AlexGilleran** [comment]
Yeah so I don't remember exactly, but every time a connector runs it changes the `sourcetag` column of the records that are created/updated - these don't generate change events. At the end of crawling it issues a command that deletes all the records that are marked as being sourced from that connector, but don't have the right tag.

When this happens either the events aren't getting created (like you said), or the events aren't getting pushed out to the indexer webhook, or the indexer is getting the events but it's not doing the right thing with them.
